modules/afamqp/afamqp.c

```diff
@@ -791,7 +791,7 @@ afamqp_dd_new(GlobalConfig *cfg)
   self->super.worker.insert = afamqp_worker_insert;
 
   self->super.format_stats_instance = afamqp_dd_format_stats_instance;
-  self->super.stats_source = SCS_AMQP;
+  self->super.stats_source = stats_register_type("amqp");
 
```

Comment:
it might actually make a lot of things simpler if we converted stats_source into a `const char *` and then pass the amqp string there and make the actual allocation in lib.

Reply:
even better idea, make SCS_AMQP equal ot the "amqp" string. with that the code does not even have to change.

Reply:
I was thinking about this version too, I even started to implement it, but there were some technical and theoretical questions that I could not reliably answer. I will give another try now.
The changeset will definitely will complex, because I need to separate the stats name and direction into separate parameters for the registration, and this will have interesting consequences. Marking the PR wip.

Reply:
In this test: https://github.com/balabit/syslog-ng/blob/f3298e8d187c5b2ac9a89755d3af79595b5553b2/lib/stats/tests/test_dynamic_ctr_reg.c#L72
we link two different names (SCS_HOST, SCS_SENDER) to the same counter. Am I right that this is invalid? 19 | 27 = 27, so I think during lookup it would print "sender", but I am not sure if this is intended behavior. Do we want to support such thing? I did not see any such application apart from test environment.

If this is a thing, then the const gchar * version cannot work. More, dynamic registration of the current version cannot not work either, because we cannot register constants randomly, we should maintain some relations like SCS_HOST | SCS_SENDER = SCS_SENDER.